ksqldb-engine/src/main/java/io/confluent/ksql/services/SandboxedSchemaRegistryClient.java

```diff
@@ -36,21 +43,193 @@
 
   static SchemaRegistryClient createProxy(final SchemaRegistryClient delegate) {
     Objects.requireNonNull(delegate, "delegate");
-
-    return LimitedProxyBuilder.forClass(SchemaRegistryClient.class)
-        .swallow("register", anyParams(), 123)
-        .swallow("getId", anyParams(), 123)
-        .forward("getAllSubjects", methodParams(), delegate)
-        .forward("getSchemaById", methodParams(int.class), delegate)
-        .forward("getLatestSchemaMetadata", methodParams(String.class), delegate)
-        .forward("getSchemaBySubjectAndId", methodParams(String.class, int.class), delegate)
-        .forward("testCompatibility",
-            methodParams(String.class, ParsedSchema.class), delegate)
-        .swallow("deleteSubject", methodParams(String.class), Collections.emptyList())
-        .forward("getVersion", methodParams(String.class, ParsedSchema.class), delegate)
-        .build();
+    return new SandboxSchemaRegistryCache(delegate);
   }
 
   private SandboxedSchemaRegistryClient() {
   }
+
+  private static final class SandboxSchemaRegistryCache implements SchemaRegistryClient {
+    private final SchemaRegistryClient delegate;
+
+    private int nextSchemaId = Integer.MAX_VALUE;
+    private final Map<String, ParsedSchema> subjectCache = new HashMap<>();
+    private final Map<String, Integer> subjectToId = new HashMap<>();
+    private final Map<Integer, ParsedSchema> idCache = new HashMap<>();
+
+    private SandboxSchemaRegistryCache(final SchemaRegistryClient delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public Optional<ParsedSchema> parseSchema(
+        final String schemaType,
+        final String schemaString,
+        final List<SchemaReference> references) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int register(final String subject, final ParsedSchema parsedSchema) {
+      if (subjectCache.containsKey(subject)) {
+        throw new IllegalStateException("Subject '" + subject + "' already in use.");
+      }
+      final int schemaId = nextSchemaId--;
+      subjectCache.put(subject, parsedSchema);
+      subjectToId.put(subject, schemaId);
+      idCache.put(schemaId, parsedSchema);
+      return schemaId;
+    }
+
+    @Override
+    public int register(
+        final String subject,
+        final ParsedSchema parsedSchema,
+        final int version,
+        final int id) {
+      return -1; // swallow
```

Comment:
Why is this swallowed whereas previous `register` does actual work?

Reply:
Old code swallows both: `.swallow("register", anyParams(), 123)`

I don't think ksqlDB actually uses this method? I just added it to make the existing unit tests pass.